// plugin-welcome/src/api/models/EntBillingstatus.ts
/* tslint:disable */
/* eslint-disable */
/**
 * SUT SA Example API
 * This is a sample server for SUT SE 2563
 *
 * The version of the OpenAPI document: 1.0
 * Contact: support@swagger.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import {
    EntBillingstatusEdges,
    EntBillingstatusEdgesFromJSON,
    EntBillingstatusEdgesFromJSONTyped,
    EntBillingstatusEdgesToJSON,
} from './';

/**
 * 
 * @export
 * @interface EntBillingstatus
 */
export interface EntBillingstatus {
    /**
     * Billingstatusname holds the value of the "billingstatusname" field.
     * @type {string}
     * @memberof EntBillingstatus
     */
    billingstatusname?: string;
    /**
     * 
     * @type {EntBillingstatusEdges}
     * @memberof EntBillingstatus
     */
    edges?: EntBillingstatusEdges;
    /**
     * ID of the ent.
     * @type {number}
     * @memberof EntBillingstatus
     */
    id?: number;
}

export function EntBillingstatusFromJSON(json: any): EntBillingstatus {
    return EntBillingstatusFromJSONTyped(json, false);
}

export function EntBillingstatusFromJSONTyped(json: any, ignoreDiscriminator: boolean): EntBillingstatus {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'billingstatusname': !exists(json, 'billingstatusname') ? undefined : json['billingstatusname'],
        'edges': !exists(json, 'edges') ? undefined : EntBillingstatusEdgesFromJSON(json['edges']),
        'id': !exists(json, 'id') ? undefined : json['id'],
    };
}

export function EntBillingstatusToJSON(value?: EntBillingstatus | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'billingstatusname': value.billingstatusname,
        'edges': EntBillingstatusEdgesToJSON(value.edges),
        'id': value.id,
    };
}
