// plugin-welcome/src/api/models/EntBillEdges.ts
/* tslint:disable */
/* eslint-disable */
/**
 * SUT SA Example API
 * This is a sample server for SUT SE 2563
 *
 * The version of the OpenAPI document: 1.0
 * Contact: support@swagger.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import {
    EntBillingstatus,
    EntBillingstatusFromJSON,
    EntBillingstatusFromJSONTyped,
    EntBillingstatusToJSON,
    EntEmployee,
    EntEmployeeFromJSON,
    EntEmployeeFromJSONTyped,
    EntEmployeeToJSON,
    EntRepairinvoice,
    EntRepairinvoiceFromJSON,
    EntRepairinvoiceFromJSONTyped,
    EntRepairinvoiceToJSON,
} from './';

/**
 * 
 * @export
 * @interface EntBillEdges
 */
export interface EntBillEdges {
    /**
     * 
     * @type {EntBillingstatus}
     * @memberof EntBillEdges
     */
    billingstatus?: EntBillingstatus;
    /**
     * 
     * @type {EntEmployee}
     * @memberof EntBillEdges
     */
    employee?: EntEmployee;
    /**
     * 
     * @type {EntRepairinvoice}
     * @memberof EntBillEdges
     */
    repairinvoice?: EntRepairinvoice;
}

export function EntBillEdgesFromJSON(json: any): EntBillEdges {
    return EntBillEdgesFromJSONTyped(json, false);
}

export function EntBillEdgesFromJSONTyped(json: any, ignoreDiscriminator: boolean): EntBillEdges {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'billingstatus': !exists(json, 'Billingstatus') ? undefined : EntBillingstatusFromJSON(json['Billingstatus']),
        'employee': !exists(json, 'Employee') ? undefined : EntEmployeeFromJSON(json['Employee']),
        'repairinvoice': !exists(json, 'Repairinvoice') ? undefined : EntRepairinvoiceFromJSON(json['Repairinvoice']),
    };
}

export function EntBillEdgesToJSON(value?: EntBillEdges | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'billingstatus': EntBillingstatusToJSON(value.billingstatus),
        'employee': EntEmployeeToJSON(value.employee),
        'repairinvoice': EntRepairinvoiceToJSON(value.repairinvoice),
    };
}
